core/server.ts

```diff
@@ -78,23 +73,15 @@ export default class Server {
   }
 
   /** Start the server */
-  async start() {
-    const { port } = this.options;
-
-    this.#server = new HttpServer({
-      port: port,
-      handler: (request, connInfo) => this.handle(request, connInfo),
-    });
-
+  start(options: Deno.ServeOptions = { port: 8000 }) {
```

Comment:
The `start` shouldn't  have any option.

Reply:
And due this class has events, probably it should handle the `Deno.Server` events automatically. I mean:

```js
options.onError = (error) => this.dispatchEvent({ type: "error", error });
options.onListen = () => this.dispatchEvent({ type: "start" });
```

Reply:
> And due this class has events, probably it should handle the `Deno.Server` events automatically. I mean:
> 
> ```js
> options.onError = (error) => this.dispatchEvent({ type: "error", error });
> options.onListen = () => this.dispatchEvent({ type: "start" });
> ```

It looks like `Deno.ServeOptions`'s `onError` needs to return a `Response`.

```ts
interface ServeOptions {
  /** The handler to invoke when route handlers throw an error. */
  onError?: (error: unknown) => Response | Promise<Response>;
  /** The callback which is called when the server starts listening. */
  onListen?: (params: { hostname: string; port: number }) => void;
}
```

Reply:
Okay, in this case, onError can be an option in the constructor. Something like `errorResponse: (error) => Response`

Reply:
So onError executes this.dispatchEvent and returns a this.errorResponse?  Is there a suitable default value?

Reply:
I don't know. Maybe no default value for now.